### ISSUE TITLE
Import XML Overwrites entire database

### ISSUE THREAD

**Issue by wwwerk** [MacPass/MacPass#1018]
**Describe the bug**
Import XML overwrites the contents of the active DB

**To Reproduce**
File > Import > Import XML

**Expected behavior**
The expected behavior for File > Import > Import XML is to append entries to the open database. Instead it wipes the entire contents of the active DB and overwrites them with the imported entries. 

**Desktop (please complete the following information):**
 - macOS: 10.13.6
 - Version 0.7 (20171024171012-continuous)

**Comment by mstarke** [comment]
The task is not that simple. Since the xml file can contain entries and groups with the sam UUID it would actually require a merge. The current behaviour is more a open than an import.

The whole import system needs a lot more work to allow for plugins as well.

My idea would be:

Import
1. Replace current content
1. Merge with current content
1. Append to current content

Another idea might be that the Replace, Merge and Append gets shown only when the import has enough information to guide the user, e.g. if a conflict was detected, merge and replace are suggested and append explains what need to be done to allow for an append, or if nothing at all is conflicting then only append or replace are possible.

**Comment by JSANL** [comment]
I mean from what I've seen there is not even a warning or anything?! Like it literally just wiped all my passwords and I did not even notice it because I thought it just opend a new database. This is really dangerous!

**Comment by mstarke** [comment]
@JSANL you are correct, it's dangerous and should be changed. I think I will get to it when I get time to finish the import/export re-structuring for plugin support.
